File: src/main/java/fr/marstech/mtlinkspray/enums/ViewNameEnum.java
package fr.marstech.mtlinkspray.enums;

import lombok.Getter;

@Getter
public enum ViewNameEnum {
    HOME("index", "Home"),
    SPRAY("spray", "Spray"),
    SHORTENER("shortener", "Shortener"),
    ABUSE("abuse", "Abuse"),
    ;

    private final String viewName;
    private final String viewPage;

    ViewNameEnum(String viewName, String viewPage) {
        this.viewName = viewName;
        this.viewPage = viewPage;
    }
}

File: src/main/java/fr/marstech/mtlinkspray/enums/ViewNameEnum.kt
package fr.marstech.mtlinkspray.enums

import lombok.Getter

@Getter
enum class ViewNameEnum(val viewName: String, val viewPage: String, val isInMenu: Boolean = true) {
    HOME("index", "Home"),
    SPRAY("spray", "Spray"),
    SHORTENER("shortener", "Shortener"),
    ABUSE("abuse", "Abuse"),
    UUID("uuid", "Uuid"),
    DASHBOARD("dashboard", "Dashboard"),
    ERROR("error", "Error", false);

    companion object {
        @JvmStatic
        fun inMenu(): List<ViewNameEnum> = entries.filter(ViewNameEnum::isInMenu)
    }
}
